app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_admin
+
+  def require_admin
+    render file: "public/404" unless current_admin?
```

Comment:
we could think about less harsh redirects... although I do like that sweet 404 page.